DylibBinder/DBParameter.cs

```diff
@@ -0,0 +1,111 @@
+		public static DBParameter CreateConstructorParameter (int index) => new DBParameter (".Type", "", "self", index, ParameterOptions.IsConstructor);
+		public static DBParameter CreateGetterParameter () => new DBParameter (null, null, null, 0, ParameterOptions.IsEmptyParameter);
+		public static DBParameter CreateSetterParameter () => new DBParameter ("", "", "self", 0, ParameterOptions.HasInstance);
+	}
+
+
+	internal class DBParameterList : IAssociatedTypes {
+		public int Index { get; }
+		public List<DBParameter> ParameterCollection { get; } = new List<DBParameter> ();
+		public DBAssociatedTypes AssociatedTypes { get; } = new DBAssociatedTypes ();
+
+		public DBParameterList (SwiftBaseFunctionType signature, bool hasInstance, bool isConstructor, int index)
+		{
+			Index = index;
+			int parameterIndex = 0;
+			AssociatedTypes.AssociatedTypeCollection.UnionWith (signature.GetAssociatedTypes ());
+
+			if (hasInstance || isConstructor) {
+				ParameterCollection.Add (hasInstance ? DBParameter.CreateInstanceParameter (parameterIndex) : DBParameter.CreateConstructorParameter (parameterIndex));
+				return;
+			}
+
+			foreach (var parameter in signature.EachParameter) {
+				var type = SwiftTypeToString.MapSwiftTypeToString (parameter);
+				var publicName = "_";
+				var privateName = "privateName";
+
+				if (parameter.Name != null)
+					publicName = privateName = parameter.Name.Name;
+
+				ParameterCollection.Add (new DBParameter (type, publicName, privateName, parameterIndex, parameter.IsVariadic ? ParameterOptions.IsVariadic : ParameterOptions.None));
+				parameterIndex++;
+			}
+		}
+
+		public DBParameterList (string propertyType, bool hasInstance, int index)
+		{
+			Index = index;
+
+			if (hasInstance) {
+				ParameterCollection.Add (DBParameter.CreateInstanceParameter ());
+				return;
+			}
+
+			if (propertyType == "Getter") {
+				ParameterCollection.Add (DBParameter.CreateGetterParameter ());
+			} else {
+				ParameterCollection.Add (DBParameter.CreateSetterParameter ());
+			}
+		}
+	}
+
+	internal class DBParameterLists : IAssociatedTypes {
+		public List<DBParameterList> ParameterListCollection { get; } = new List<DBParameterList> ();
+		public DBAssociatedTypes AssociatedTypes { get; } = new DBAssociatedTypes ();
+
+		public DBParameterLists (SwiftBaseFunctionType signature, bool hasInstance)
+		{
+			var parameterListIndex = 0;
+			if (hasInstance) {
+				ParameterListCollection.Add (new DBParameterList (signature, true, false, parameterListIndex));
+				parameterListIndex++;
+			} else if (signature.IsConstructor) {
+				ParameterListCollection.Add (new DBParameterList (signature, false, true, parameterListIndex));
+				parameterListIndex++;
+			}
+
+			ParameterListCollection.Add (new DBParameterList (signature, false, false, parameterListIndex));
+			AssociatedTypes.AssociatedTypeCollection.UnionWith (ParameterListCollection.GetChildrenAssociatedTypes ());
```

Comment:
Is this suppose to execute always or as an 'else' for the other two branches of the if?

Reply:
These should always happen!